src/types/datetime.rs

```diff
@@ -289,11 +348,41 @@ impl PyDateTime {
                 microsecond as c_int,
                 opt_to_pyobj(tzinfo),
                 api.DateTimeType,
-            );
-            py.from_owned_ptr_or_err(ptr)
+            )
+            .assume_owned_or_err(py)
+            .downcast_into_unchecked()
         }
     }
 
+    /// Deprecated form of [`PyDateTime::new_bound_with_fold`].
+    #[allow(clippy::too_many_arguments)]
+    pub fn new_with_fold<'p>(
```

Comment:
This misses the deprecation (or was it intentional?)

Reply:
Just a typical case of my error-prone typing, good catch & will fix!